src/python/base/external_tasks.py

```diff
@@ -0,0 +1,62 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""External tasks."""
+
+from datastore import data_handler
+from google_cloud_utils import blobs
+from google_cloud_utils import pubsub
+from metrics import logs
+from system import environment
+
+
+def add_external_task(command, testcase_id, job):
+  """Add external task."""
+  if command != 'progression':
+    # Only progression is supported.
+    return
+
+  pubsub_client = pubsub.PubSubClient()
+  topic_name = job.external_reproduction_topic
+  assert topic_name is not None
+
+  testcase = data_handler.get_testcase_by_id(testcase_id)
+  fuzz_target = testcase.get_fuzz_target()
+
+  memory_tool_name = environment.get_memory_tool_name(job.name)
+  sanitizer = environment.SANITIZER_NAME_MAP.get(memory_tool_name)
+  job_environment = job.get_environment()
+  build_path = (
```

Comment:
Do we need a bail out for custom binaeries.

Reply:
This should never happen (it doesn't make sense for an "external" infra to know about custom binaries). I can add an exception here.